utils_test.go

```diff
@@ -31,3 +34,77 @@ func TestGenerateEventName(t *testing.T) {
 		t.Errorf("Test should failed as event name length is too big for the kernel and free space for function Name is < %d", MinFunctionNameLen)
 	}
 }
+
+// return "/usr/lib/ld-2.33.so" for example
+func detectLDLoaderPath(t *testing.T) string {
+	out, err := exec.Command("bash", "-c", `grep '/ld-.*\.so$' /proc/self/maps | head -n1 | awk '{print $6}'`).Output()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return string(out[:len(out)-1])
+}
+
+func openAndListSymbols(t *testing.T, path string) []elf.Symbol {
+	f, sym, err := OpenAndListSymbols(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	return sym
+}
+
+func openAndListSymbolsFromPID(t *testing.T, path string) []elf.Symbol {
+	f, sym, err := OpenAndListSymbolsFromPID(os.Getpid(), path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	return sym
+}
+
+func TestOpenAndListSymbols(t *testing.T) {
+	path := detectLDLoaderPath(t)
+
+	syms := openAndListSymbols(t, path)
+	t.Log("NR symbol loaded:", len(syms))
```

Comment:
Can we add some asserts to this test?

Reply:
wasn't possible to failed (ie t.Fatal() in the caller) but I move to a more coventional testing/assert using stretchr/testify.NoError()